agent/php_nrini.c

```diff
@@ -222,6 +223,53 @@
 
 ZEND_DECLARE_MODULE_GLOBALS(newrelic)
 
+#define NR_INI_PREFIX_LEN 9
+
+char* nr_ini_to_env(const char* ini_name) {
+  char* env_name = NULL;
+  char* buf = NULL;
+  char* ini_upper = NULL;
+  const char* NR_PREFIX_STR = "NEW_RELIC";
+  int ini_len = nr_strlen(ini_name);
+
+  // 'newrelic.' is 9 characters - anything less should be rejected.
+  if (NULL == ini_name || nr_strlen(ini_name) <= NR_INI_PREFIX_LEN) {
+    return NULL;
+  }
+
+  // ini value should start with the 'newrelic.' prefix
+  if (0 != nr_stridx(ini_name, "newrelic.")) {
+    return NULL;
+  }
+
+  // algorithm:
```

Comment:
There is an additional requirement that consecutive underscores should be merged into a single underscore.  I don't think this function would handle that case.

Reply:
reworked the algorithm to handle this edge case here: d58b00e68f8ab2587fa3cfee4d4cde5b64f44e88